CodingTracker.AlvinGraham/CodingTracker.AlvinGraham/UserInterface.cs

```diff
@@ -0,0 +1,183 @@
+
+			var userChoice = AnsiConsole.Prompt(selectionPrompt);
+
+			try
+			{
+				switch (userChoice)
+				{
+					case MainMenuChoices.AddRecord:
+						Utilities.ClearScreen("Adding Record");
+						AddRecord();
+						break;
+					case MainMenuChoices.ViewRecords:
+						Utilities.ClearScreen("Viewing Records");
+						var dataAccess = new DataAccess();
+						var records = dataAccess.GetAllRecords();
+						ViewRecords(records);
+						break;
+					case MainMenuChoices.UpdateRecord:
+						Utilities.ClearScreen("Updating Record");
+						UpdateRecord();
+						break;
+					case MainMenuChoices.DeleteRecord:
+						Utilities.ClearScreen("Deleting Record");
+						DeleteRecord();
+						break;
+					case MainMenuChoices.LiveTrack:
+						Utilities.ClearScreen("Recoring Live Tracking Session");
+						LiveTrack();
+						break;
+					case MainMenuChoices.FilterRecords:
+						FilterRecords();
+						break;
+					case MainMenuChoices.Reports:
+						Utilities.ClearScreen("Reports Menu");
+						RunReports();
+						break;
+					case MainMenuChoices.ManageGoals:
+						Utilities.ClearScreen("Managing Goals");
+						ManageGoals();
+						break;
+					case MainMenuChoices.Quit:
+						Console.WriteLine("Goodbye");
+						isMenuRunning = false;
+						break;
+				}
+			}
+			catch (ArgumentException)
+			{
+				Console.WriteLine("\nReturning to Main Menu.  Press any key to continue...");
+				Console.ReadKey();
+			}
+		}
+	}
+
+	private static void ManageGoals()
+	{
+		var goalsSession = new Goals();
+		goalsSession.GoalsMenu();
+	}
+
+	private static void RunReports()
+	{
+		var reportSession = new Reports();
+		reportSession.RunReports();
+		Console.WriteLine("\nPress any Key to Return to main menu.");
+		Console.ReadKey();
+	}
+
+	private static void FilterRecords()
+	{
+		var filterSession = new RecordFilter();
+		filterSession.FilterRecordsMenu();
+		Console.WriteLine("\nPress any Key to Return to main menu.");
+		Console.ReadKey();
+	}
+
+	private static void LiveTrack()
+	{
+		var trackingSession = new LiveTracker();
+		trackingSession.TrackSession();
+		Console.WriteLine("\nPress any Key to Return to main menu.");
+		Console.ReadKey();
+	}
+
+	private static void DeleteRecord()
+	{
+		var dataAccess = new DataAccess();
+		var records = dataAccess.GetAllRecords();
+		ViewRecords(records);
+
+		var id = Utilities.GetNumber("Please type the id of the habit you want to delete.");
+
+		if (!AnsiConsole.Confirm("Are you sure?"))
+			return;
+
+		var response = dataAccess.DeleteRecord(id);
+
+		var responseMessage = response < 1
+			? $"Record with id {id} doesn't exit. Press any key to return to Main Menu."
+			: "Record deleted successfully. Press any key to return to Main Menu";
+
+		Console.WriteLine(responseMessage);
+		Console.ReadKey();
+	}
+
+	private static void UpdateRecord()
+	{
+		var dataAccess = new DataAccess();
+		var records = dataAccess.GetAllRecords();
+		ViewRecords(records);
```

Comment:
It will be nice not to show an empty table, especially when updating/Deleting a record.
You can return when there are no records in database.

Reply:
Acknowledged. I made the changes for Updating/Deleting records and updating/deleting goals.